### PR TITLE
odin 2024-10

### PR DESCRIPTION
**Pull request by laytan**
Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> /home/linuxbrew/.linuxbrew/Cellar/odin/2024-10/bin/odin run glfw.odin -file
  /tmp/odin-test-20241001-19716-u74zzi/glfw: error while loading shared libraries: libglfw.so.3: cannot open shared object file: No such file or directory
```

File: Formula/o/odin.rb [hunk 1]
```diff
@@ -56,7 +55,13 @@ def install
...
     end
 
-    ln_s Formula["glfw"].lib/"libglfw3.a", buildpath/"vendor/glfw/lib/darwin/libglfw3.a"
+    glfw_installpath = if OS.linux?
+      "vendor/glfw/lib"
+    else
+      "vendor/glfw/lib/darwin"
+    end
+
+    ln_s Formula["glfw"].lib/"libglfw3.a", buildpath/glfw_installpath/"libglfw3.a"
```
Comment (carlocab) [inline Formula/o/odin.rb:64]: Does this symlink not get broken once we install the `vendor` directory into the `prefix`?

Comment (laytan) [reply Formula/o/odin.rb:64]: The test I added that passes `-define:GLFW_SHARED=false` should break if that is the case I think, that one would try to find and link with it. I also tried it locally and it seems to work.

Let me know if there's something else you want to see here or if there's something else I could test to make sure.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**